tests/unit/src/test/scala/tests/TypeDefinitionLspSuite.scala

```diff
@@ -0,0 +1,146 @@
+package tests
+
+import org.eclipse.{lsp4j => l}
+import scala.concurrent.Future
+import scala.concurrent.duration.Duration
+import scala.meta.internal.metals.MetalsServerConfig
+import scala.meta.internal.metals.StatisticsConfig
+
+object TypeDefinitionLspSuite
+    extends BaseLspSuite("typeDefinition")
+    with TestHovers {
+  override def serverConfig: MetalsServerConfig =
+    super.serverConfig.copy(
+      statistics = new StatisticsConfig("diagnostics")
+    )
+
+  override def testAsync(
+      name: String,
+      maxDuration: Duration = Duration("3min")
+  )(run: => Future[Unit]): Unit = {
+    if (isWindows) {
+      // src.zip is missing on Appveyor which breaks definition tests.
+      ignore(name) {}
+    } else {
+      super.testAsync(name, maxDuration)(run)
+    }
+  }
+
+  def locationFromString(str: String): l.Location = {
+    def positionFromString(str: String): l.Position = {
+      val ints = str.split(":").map(_.toInt)
+      new l.Position(ints.head, ints.last)
+    }
+
+    val split = str.split("""/\\""")
+    val (uri, rest) = (split.head, split.tail)
+    val range =
+      new l.Range(positionFromString(rest.head), positionFromString(rest.last))
+    new l.Location(
+      workspace.toURI.resolve(uri).toString.replaceFirst("/", "///"),
+      range
+    )
+  }
+
+  def createLocation(
+      uri: String,
+      begin: (Int, Int),
+      end: (Int, Int)
+  ): l.Location = {
+    val startPos = new l.Position(begin._1, begin._2)
+    val endPos = new l.Position(end._1, end._2)
+
+    new l.Location(uri, new l.Range(startPos, endPos))
+  }
+
+  testAsync("int") {
+    for {
+      _ <- server.initialize(
+        """/metals.json
+          |{
+          |   "a":{}
+          |}
+        """.stripMargin
+      )
+      _ <- server.assertTypeDefinition(
+        "a/src/main/scala/a/Main.scala",
+        """
+          |object Main {
+          |  val ts@@t: Int = 2
+          |}""".stripMargin,
+        locationFromString(".metals/readonly/scala/Int.scala/\\25:21/\\25:24")
+      )
+    } yield ()
+  }
+
+  testAsync("basic") {
+    cleanWorkspace()
+    for {
+      _ <- server.initialize(
```

Comment:
We can move initialize to separate helper method like in ImplementationProvider.